### PR TITLE
Filter Unit Tests, Documentation Improvements, & 2D Action Classifier

### PR DESCRIPTION
**Pull request by pgluss**
Created unit tests for filters such that everything has at minimum 70% coverage, updated the README and commented a few functions to begin building a more in depth series of function comments throughout the code base. 

In addition, added a UDF for a 2D action classifier and the initial outline of a respective unit test.

### CODE REVIEW

**Comment by pgluss** [comment]
It looks like the Travis CI failed due to an environment issue. I'll update environment_mac.yml so that it includes what's in environment.yml

File: src/udfs/video_action_classification.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+from typing import List, Tuple
+from glob import glob
+import numpy as np
+import random
+import os
+
+
+class ActionClassificationLoader:
+    def __init__(self, path):
```
Comment (gaurav274) [inline src/udfs/video_action_classification.py]: Ideally, the loaders shouldn't be in UDFs. We can remove this from here and maybe use AbstractVideoLoader already defined in storage.

File: test/filters/test_kdewrapper.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+from src.filters.kdewrapper import KernelDensityWrapper
```
Comment (gaurav274) [inline test/filters/test_kdewrapper.py]: make a subclass of unittest class. see other test cases

File: test/filters/test_kdewrapper.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+from src.filters.kdewrapper import KernelDensityWrapper
+import numpy as np
+
+
+def test_KD_Wrapper():
```
Comment (gaurav274) [inline test/filters/test_kdewrapper.py]: make a subclass of unittest class. see other test cases

File: test/filters/test_minimum_filter.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
+from src.filters.minimum_filter import FilterMinimum
+from src.filters.models.ml_pca import MLPCA
+from src.filters.models.ml_dnn import MLMLP
+import numpy as np
+
+
+def test_FilterMinimum():
+    # Construct the filter minimum and test it with randomized values
+    # The idea is just to run it and make sure that things run to completion
+    # No actual output or known inputs are tested
```
Comment (gaurav274) [inline test/filters/test_minimum_filter.py]: make a subclass of unittest class. see other test cases

File: test/filters/test_pp.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import numpy as np
+from src.filters.pp import PP
+
+
+def test_PP():
```
Comment (gaurav274) [inline test/filters/test_pp.py]: make a subclass of unittest class. see other test cases

**Comment by pgluss** [comment]
Requested changes were fixed, however a previous merge on upstream unrelated to me or this PR is causing CI to fail.

**Pull request merged**